catalyst-gateway/bin/src/service/utilities/middleware/tracing_mw.rs

```diff
@@ -276,14 +277,21 @@ impl<E: Endpoint> Endpoint for TracingEndpoint<E> {
                     //};
                     let panic: Option<Uuid> = None;
 
-                    // Get the error message, and also for now, log the error to ensure we are
-                    // preserving all data.
-                    error!(err = ?err, "HTTP Response Error:");
-                    // Only way I can see to get the message, may not be perfect.
-                    let error_message = err.to_string();
-
                     // Convert the error into a response, so we can deal with the error
+                    let error_message = err.to_string();
                     let resp = err.into_response();
+                    let status = resp.status();
+
+                    // Log 404 as warning, if env set
+                    if status == StatusCode::NOT_FOUND {
+                        if env::var("LOG_NOT_FOUND").is_ok() {
```

Comment:
move this `LOG_NOT_FOUND` env var definition under the `EnvVars`  structure inside `settings.rs`